doc/Language/syntax.pod6

```diff
@@ -223,109 +223,193 @@ say 'code again';
-add I<symbols> within angle brackets. This is a mechanism that supports
-interpolation; which is also supported by colonpairs if the interpolated
-variable is inserted in parentheses. Please note that resolution of
-names often happens at compile time, so interpolation values must be
-known at compile time.
-
-    my $foo:bar<2> = 2;
-    constant $c = 42;
-    my $a:foo<42> = "answer";
-    say $a:foo«$c»;    # OUTPUT: «answer␤»
-    my $buz = "qux";
-    my $bur::quux = 7;
-    say $bur::($buz);  # OUTPUT: «7␤»
+    infix:<+>
+    infix:<<+>>
+    infix:«+»
+    infix:['+']
+    infix:('+')
 
-The first statement is an example of I<X<extended identifiers>>. Extended
-identifiers are used extensively in the definition of the language, for
-instance, to define L<terms|language/syntax#Term_term:%3C%3E>. They include a
-colon C<:> in their name, and might include a bracketing-quoting construct such
-as C<«»>, C«<>» or C<['']>. These bracketing constructs are interchangeable. For
-instance
+Similarly, all of this works:
 
     my $foo:bar<baz> = 'quux';
-    say $foo:bar«baz»; # OUTPUT: «quux␤»
+    say $foo:bar«baz»;                                  # OUTPUT: «quux␤»
     my $take-me:<home> = 'When the glory has no end';
-    say $take-me:['home'];# OUTPUT: «When the glory has no end␤»
+    say $take-me:['home'];                              # OUTPUT: «When the glory has no end␤»
+    my $foo:bar<2> = 5;
+    say $foo:bar(1+1);                                  # OUTPUT: «5␤»
+
+Where an extended identifier comprises two or more colon pairs, their order
+is generally significant:
 
-This extended syntax is used, for intance, for module versions and
-authors in the shape:
-C«HTTP::UserAgent:ver<1.1.16>:auth<github:sergot>».
+    my $a:b<c>:d<e> = 100;
+    my $a:d<e>:b<c> = 200;
+    say $a:b<c>:d<e>;               # OUTPUT: «100␤», NOT: «200␤»
 
-Colonpairs with empty key (e.g. C«:<foo>») aer reserved in subroutine
-names, as they're used to introduce
-L<a new custom operator|/language/functions#Defining_operators> into
-a particular category. Starting with 6.d language, colonpairs with
-C<sym> key (e.g. C«:sym<foo>») are reserved for possible future use.
+An exception to this rule is I<module versioning>; so these identifiers
+effectively name the same module:
 
-There are slight differences between the three constructs: C«<>» cannot be used
-for interpolation of constant names.
+    use ThatModule:auth<Somebody>:ver<2.7.18.28.18>
+    use ThatModule:ver<2.7.18.28.18>:auth<Somebody>
 
-    constant $what= 'are';
+Furthermore, the adverbial pair syntax in extended identifiers supports
+compile-time interpolation, which mandates the use of
+L<constants|/language/terms#Constants> for the interpolation values:
+
+    constant $c = 42;               # Constant binds to Int; $-sigil enables interpolation
+    my $a:foo<42> = "answer";
+    say $a:foo«$c»;                 # OUTPUT: «answer␤»
+
```

Comment:
I know this is exactly how it was in the original, but maybe, and as long as we're trying to establish some best practice for identifiers here, it would be better to define the constant as \c to differentiate it from run-time variables.

Reply:
Can you provide a working example? I can't get the interpolation to work with a sigilless "variable"; i.e.

 constant \c = 42;
 my $a:foo<42> = "answer";
 say $a:foo«c»;

does not compile. I assume the sigil is necessary to perform the interpolation, which is one of the reasons for having sigils in the first place, so that makes sense. It is also why the proposed text (in the example) states that the $-sigil enables interpolation.